Source-Code/FlareDraw.cs

```diff
@@ -192,9 +192,22 @@ private void GenerateBodyFlares()
             //ListChildren(sm.systemPrefab.rootBody, 0);
             //--- HACK--
 
+            //If Kerbin is parented to the Sun, set its SMA- otherwise iterate
+            //through celestial bodies to locate which is parented to the Sun
+            //and has Kerbin as a child. Set the highest parents SMA to kerbinSMA
             if(BodyFlare.kerbinSMA <= 0.0)
             {
-                BodyFlare.kerbinSMA = FlightGlobals.Bodies[1].orbit.semiMajorAxis;
+                if(FlightGlobals.Bodies[1].referenceBody == FlightGlobals.Bodies[0])
+                    BodyFlare.kerbinSMA = FlightGlobals.Bodies[1].orbit.semiMajorAxis;
+                else
+                {                    
+                    foreach(CelestialBody current in FlightGlobals.Bodies)
```

Comment:
Probably, you could then cutout the check to make sure it wasn't trying to
look at the parent of the Sun (Bodies[0]), which pulls out a NRE, and also
skip going over Kerbin. Could change it to:

else
{
     for(int = 2; i < FlightGlobals.Bodies.Count; i++)
     {
          if(FlightGlobals.Bodies[i].referenceBody == FlightGlobals.Bodies[0]
&& FlightGlobals.Bodies[i].HasChild(FlightGlobals.Bodies[1])
          {
               BodyFlare.kerbinSMA =
FlightGlobals.Bodies[i].orbit.semiMajorAxis;
               break;
          }
     }
}

This is what I initially tried setting up, so it would stop looking after
it found the right parent. I don't think I was seeing "Count" as an
autocomplete option and had tried messing around with "Max" before changing
to a "foreach" statement. Going up the chain might not be a bad idea either.

else
{
     CelestialBody temp = FlightGlobals.Bodies[1].referenceBody;

     while(BodyFlare.kerbinSMA == -1)
     {
          if(temp.referenceBody == FlightGlobals.Bodies[0])
               BodyFlare.kerbinSMA = temp.orbit.semiMajorAxis;
          else
               temp = temp.referenceBody;
     }
}

Again, I'm not sure if there would be a scenario that had Kerbin double
parented or if you would want to account for it. Just doing an "else" and
taking the orbit from Kerbin's parent sets everything up as needed for New
Horizons. It has been years since I've done much coding, working on getting
some of the rust off.


On Thu, Aug 20, 2015 at 10:11 AM, MOARdV <notifications@github.com> wrote:

> In Source-Code/FlareDraw.cs
> <https://github.com/MOARdV/DistantObject/pull/20#discussion_r37533194>:
>
> >              if(BodyFlare.kerbinSMA <= 0.0)
> >              {
> > -                BodyFlare.kerbinSMA = FlightGlobals.Bodies[1].orbit.semiMajorAxis;
> > +                if(FlightGlobals.Bodies[1].referenceBody == FlightGlobals.Bodies[0])
> > +                    BodyFlare.kerbinSMA = FlightGlobals.Bodies[1].orbit.semiMajorAxis;
> > +                else
> > +                {
> > +                    foreach(CelestialBody current in FlightGlobals.Bodies)
>
> Okay. Not a problem.
>
> I actually wanted to use a for() statement that broke out when it was
> located but wasn't sure how to find how many bodies were in the index for
> the range.
>
> You mean FlightGlobals.Bodies.Count?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/MOARdV/DistantObject/pull/20/files#r37533194>.
>